metrics: Add the `PD_cluster_unhealthy_tikv_nums` alert (#11013)
pingcap/docs: alert-rules.md

    * If you confirm that the TiKV/TiFlash instance cannot be recovered, you can make it offline.
    * If you confirm that the TiKV/TiFlash instance can be recovered, but not in the short term, you can consider increasing the value of `max-down-time`. It will prevent the TiKV/TiFlash instance from being considered as irrecoverable and the data from being removed from the TiKV/TiFlash.

> #### `PD_cluster_unhealthy_tikv_nums`

Reviewer: Should we change to store instead of TiKV? Because `store_unhealth_count` is the sum of unhealthy TiKV and TiFlash.

Author: The original PR ( https://github.com/tikv/pd/pull/5612 ) was based on the [`pd.rules.yml`](https://github.com/pingcap/monitoring/blob/master/monitor-snapshot/master/operator/rules/pd.rules.yml) that is in the pingcap/monitoring repo. That file has a `PD_cluster_offline_tikv_nums`.

After applying this to the `pd.rules.yml` in this repo I failed to check if the naming is still consistent.

Note that 62c6647e29cb4251089c5a6e04a27fd991b714f0 changed the name.

I do think changing the name makes sense, but this PR is about the documentation. I think we can continue with this PR and then update the `pingcap/pd` repo and afterwards update the `docs` repo if needed.

I'll also file a PR to remove the old `pd.rules.yml` and similar files to avoid this from happening again.

Author: That might be a good idea, but should then be applied to other rules as well:
```
$ grep -E -A1 'alert:.*tikv_nums' metrics/alertmanager/pd.rules.yml
  - alert: PD_cluster_down_tikv_nums
    expr: (sum(pd_cluster_status{type="store_down_count"}) by (instance) > 0) and (sum(etcd_server_is_leader) by (instance) > 0)
--
  - alert: PD_cluster_lost_connect_tikv_nums
    expr: (sum(pd_cluster_status{type="store_disconnected_count"}) by (instance) > 0) and (sum(etcd_server_is_leader) by (instance) > 0)
--
  - alert: PD_cluster_unhealthy_tikv_nums
    expr: (sum ( pd_cluster_status{type="store_unhealth_count"} ) by (instance)
--
  - alert: PD_cluster_slow_tikv_nums
    expr: (sum(pd_cluster_status{type="store_slow_count"}) by (instance) > 0) and (sum(etcd_server_is_leader) by (instance) > 0)
```

* Alert rule:

>     `(sum ( pd_cluster_status{type="store_unhealth_count"} ) by (instance) > 0) and (sum(etcd_server_is_leader) by (instance) > 0)`

Reviewer 2: ```suggestion
    `(sum(pd_cluster_status{type="store_unhealth_count"}) by (instance) > 0) and (sum(etcd_server_is_leader) by (instance) > 0)`
```

* Description:

>     Indicates that there are unhealthy stores. If the situation persists for some time (configurable, by default 30m) the store state is likely to change to down and trigger the [`PD_cluster_down_store_nums`](#PD_cluster_down_store_nums) alert.

Reviewer 2: ```suggestion
    Indicates that there are unhealthy stores. If the situation persists for some time (configured by [`max-store-down-time`](/pd-configuration-file.md#max-store-down-time), by default `30m`) the store state is likely to change to down and trigger the [`PD_cluster_down_store_nums`](#PD_cluster_down_store_nums) alert.
```

>     Indicates that there are unhealthy stores. If the situation persists for some time (configured by [`max-store-down-time`](/pd-configuration-file.md#max-store-down-time), by default `30m`) the store state is likely to change to down and trigger the [`PD_cluster_down_store_nums`](#pd_cluster_down_store_nums) alert.

Reviewer 2: ```suggestion
    Indicates that there are unhealthy stores. If the situation persists for some time (configured by [`max-store-down-time`](/pd-configuration-file.md#max-store-down-time), by default `30m`), the store state is likely to change to `Offline` and trigger the [`PD_cluster_down_store_nums`](#pd_cluster_down_store_nums) alert.
```

>     Indicates that there are unhealthy stores. If the situation persists for some time (configured by [`max-store-down-time`](/pd-configuration-file.md#max-store-down-time), by default `30m`), the store state is likely to change to `Offline` and trigger the [`PD_cluster_down_store_nums`](#pd_cluster_down_store_nums) alert.

Reviewer 3: ```suggestion
    Indicates that there are unhealthy stores. If the situation persists for some time (configured by [`max-store-down-time`](/pd-configuration-file.md#max-store-down-time), defaults to `30m`), the store is likely to change to `Offline` state, which triggers the [`PD_cluster_down_store_nums`](#pd_cluster_down_store_nums) alert.
```

* Solution:

>     * Check the status of the TiKV stores.

Reviewer 2: ```suggestion
    Check the state of the TiKV stores.
```
